tests/video/test_video.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# vim: ai ts=4 sts=4 et sw=4 nu
+
+import pathlib
+import subprocess
+import humanfriendly
+import shutil
+import pytest
+
+from zimscraperlib.download import save_large_file
+
+
+def get_video_info(src_path):
+    args = [
+        "ffprobe",
+        "-i",
+        f"file:{src_path}",
+        "-show_entries",
+        "stream=codec_name",
+        "-show_entries",
+        "format=duration,bit_rate",
+        "-v",
+        "quiet",
+        "-of",
+        "csv",
+    ]
+    print(" ".join(args))
+    ffprobe = subprocess.run(
+        args,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        universal_newlines=True,
+        check=False,
+    )
+    ffprobe_result = ffprobe.stdout.strip().split("\n")
+    streams = ffprobe_result[:-1]
+    codecs = [stream.split(",")[-1] for stream in streams]
+    format_info = ffprobe_result[-1].split(",")[1:]
+    vid_info = {
+        "codecs": codecs,
+        "duration": int(format_info[0].split(".")[0]),
+        "bitrate": int(format_info[1]),
+    }
+    return vid_info
+
+
+def test_vidcompressioncfg_defaults(vidcompressioncfg):
+    assert vidcompressioncfg.max_video_bitrate == "300k"
+    assert vidcompressioncfg.min_video_bitrate is None
+    assert vidcompressioncfg.target_video_bitrate == "300k"
+    assert vidcompressioncfg.buffersize == "1000k"
+    assert vidcompressioncfg.audio_sampling_rate == 44100
+    assert vidcompressioncfg.target_audio_bitrate == "128k"
+    assert vidcompressioncfg.ffmpeg_video_scale == "480:trunc(ow/a/2)*2"
+    qmin, qmax = vidcompressioncfg.quality_range
+    assert qmin == 30
+    assert qmax == 42
+
+
+def test_vidcompressioncfg_update(vidcompressioncfg):
+    updates = {
+        "min_video_bitrate": "300k",
+        "quality_range": (25, 35),
+        "audio_sampling_rate": 50000,
+    }
+    vidcompressioncfg.update(**updates)
+    for k, v in updates.items():
+        assert getattr(vidcompressioncfg, k) == v
+
+
+def test_ffmpeg_args(vidcompressioncfg):
+    vidcompressioncfg.update(min_video_bitrate="300k")
+    ffargs = vidcompressioncfg.to_ffmpeg_args(
+        video_codec="h264",
+        audio_codec="aac",
+        extra_ffmpeg_params=["-movflags", "+faststart"],
+    )
+    assert len(ffargs) == 32
+    options_map = [
+        ("maxrate", "max_video_bitrate"),
+        ("b:v", "target_video_bitrate"),
+        ("bufsize", "buffersize"),
+        ("ar", "audio_sampling_rate"),
+        ("b:a", "target_audio_bitrate"),
+        ("quality", "quality"),
+        ("minrate", "min_video_bitrate"),
+    ]
+    for index, option_data in enumerate(options_map):
+        option, attr = option_data
```

Comment:
you can replace that with `for index, (option, attr) in enumerate(options_map):`